src/libraries/Common/src/System/Security/Cryptography/RsaPaddingProcessor.cs

```diff
@@ -162,7 +162,6 @@ internal static RsaPaddingProcessor OpenProcessor(HashAlgorithmName hashAlgorith
 
             if (source.Length != expectedLength)
             {
-                Debug.Fail($"Provided has is {source.Length} bytes, should be {expectedLength}");
```

Comment:
The caller should be assuring this already...

Reply:
Well currently it's not. The negative test with a mismatched hash length was crashing here. Is it ok to throw here or do we need to check this earlier?